src/Aspire.Hosting/ApplicationModel/ParameterResource.cs

```diff
@@ -75,7 +75,7 @@ internal string ConfigurationKey
     /// <summary>
     /// A task completion source that can be used to wait for the value of the parameter to be set.
     /// </summary>
-    internal TaskCompletionSource<string>? WaitForValueTcs { get; set; }
+    public TaskCompletionSource<string>? WaitForValueTcs { get; set; }
```

Comment:
Making this public feels like an ick to me. Another option I considered is adding an explicit `SetValueAsync` method to this type. Publish mode is interesting since we can't asynchronously resolve each parameter like we do in the dashboard.

Reply:
You also need to make CreateInput public.

Reply:
> You also need to make CreateInput public.

Done in https://github.com/dotnet/aspire/pull/11160/commits/d1c5c7c7c7f9fee5b929882b311b6bb8a18e9b64